### ISSUE TITLE
Crop and rotate - straight line - rotated by invalid angle - DT 3.4.1

### ISSUE THREAD

**Issue by Vity01** [darktable-org/darktable#7149]
1. Open image you want to rotate in Darkroom (Crop and rotate module)
2. Use a right mouse click to drag line from one side to another - also use a mouse scroll button to zoom a target area in parallel
3. Release right mouse button to select second point

Now you see the image is not straight, rotated by weird angle. It does happen only if you use the mouse scroll. 
It looks like the Angle toggle is influenced by zooming (it takes a focus sometimes?).

DT 3.2.1
Windows 10
Click on the image to see animated GIF
![darktable-bug](https://user-images.githubusercontent.com/2512959/100740498-1a792980-33d9-11eb-9254-e6711e9fdc41.gif)

**Additional context**
Sometimes this bug behavior happens only 1x for the photo. But it happens again for another photo.

**Comment by TurboGit** [comment]
I'm surprised, this has always been working for me since many version, so there is something else. What other module are you using?

**Comment by Vity01** [comment]
@TurboGit No other or special modules. See the empty history stack. It's a first step I do with the image.

**Comment by TurboGit** [comment]
Do you reproduce that with all images ? Is that a scan ? Maybe a specific orientation in the orientation module ?

**Comment by Vity01** [comment]
Yes, I can reproduce it with all my images. Yes , it is a scan. Usually after 3rd attempt (with several undos) with the rotation, it starts to behave correctly.

**Comment by chhil** [comment]
What format is the scanned file in?

**Comment by Vity01** [comment]
@chhil PNG

**Comment by chhil** [comment]
I think it would be helpful if a file that you are having a problem with can be shared here so that the devs who work on DT can assist in solving it.

**Comment by Vity01** [comment]
The files are quite huge, I uploaded 1 one of them. The link will work 14 days.
 https://www.uschovna.cz/zasilka/FMAZ3AUAGEZWP4A5-NH3/

**Comment by ptilopteri** [comment]
that's a png, where is the raw file

**Comment by Vity01** [comment]
@ptilopteri please read the comments above, there is no RAW file.

**Comment by ptilopteri** [comment]
and I have no problem rotating the file even using the scroll wheel 
while dragging the mouse pointer to set the horizontal parameter.
I did not try vertical.

**Comment by ptilopteri** [comment]
ok, but still no problem straightening the image

**Comment by ptilopteri** [comment]
but could be a windows thing, I am on linux.

**Comment by chhil** [comment]
Reproducible on current master with shared file or any raw file.

What happens is the angle changes on zooming. You have drawn the line to straighten, now scroll to zoom in considerably, the angle visible from the end of the line changes which should have remained constant.

**Comment by chhil** [comment]
@Vity01 , your current workaround would be to adjust it without zooming by using the grid option and adding additional horizontal lines to assist with straightening.

**Comment by ptilopteri** [comment]
another work-a-round
I set short-cut keys lowercase q/w to do 1/2 degree rotation left/right
and often have no need of using the line for adjustment

**Comment by Vity01** [comment]
@chhil yes, without zooming it's working (but it's not so precise). I am not sure what you mean by adding additional horizontal line.

**Comment by chhil** [comment]
@Vity01 
![image](https://user-images.githubusercontent.com/1119439/100752706-e6b8f680-340e-11eb-9ed3-8de85e3939b6.png)

**Comment by MStraeten** [comment]
it seems the "fit" zoom moves the image a little bit. if you start in a already zoomed in scale the angle keeps constant while further scaling in or out - until you reach the "fit" scale.

**Comment by Vity01** [comment]
I would like to raise this issue again. It still happens on 3.4.1 :-(

**Comment by johnny-bit** [comment]
@Vity01 - I tried to reproduce but don't have good reproducers that would show if the issue is there or not. And if it is still there I think on my samples it's not severe enough.

There were tons of changes to crop and rotate module during dev cycle for 3.6 release. Can you try and see if with 3.6 you still have the issue (3.6 will be officially released in ~5 days)

**Comment by Vity01** [comment]
@johnny-bit I would like to, but I don't see 3.6 build for Windows anywhere to download.

**Comment by johnny-bit** [comment]
Ok, then let's wait till the weekend. darktable 3.6 official release is on 03 July 2021 planned.

**Comment by Vity01** [comment]
@johnny-bit I finally got some time to test it and the problem persists on 3.6 as well. I made a new video for you. 
There is definitely some enhancement, I was not able to reproduce it for each attempt, but I am still able to reproduce it. I think it relates to the zoom level from which I start to use the straight line. 
Note the angle information - around 1° near by line and the final degree angle in the the module info panel is different.
![ezgif com-gif-maker](https://user-images.githubusercontent.com/2512959/127478706-eab73a2a-6a87-4b88-8e44-c60b4593e105.gif)

**Comment by Vity01** [comment]
Any progress on this?

**Comment by elstoc** [comment]
We are approaching a time when "crop & rotate" will be deprecated. The rotation functionality is now part of the "rotation and perspective" module so this should probably be retested there.

**Comment by ptilopteri** [comment]
I generally utilize keyboard shortcuts to rotate +/- 0.25°
but the few times I used <rt><mouse> drag worked as expected

**Comment by elstoc** [comment]
same here, but then I've not really had issues in crop & rotate either

**Comment by Nilvus** [comment]
As crop and rotate is now deprecated for 3.8 and a new rotation and perspective is added to replace it, I close this issue. @Vity01: I think the best would be that you test that new module when 3.8 will be out (for Christmas) and if the issue is still there, it will be better to post a new issue. Of course, if you could already test actual master (future 3.8), it will be better as this could provide better feedback for 3.8 release.

**Comment by elstoc** [comment]
Agreed - I still can't reproduce this on 3.8 in rotate & perspective
